dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/IdentifiableObjectManager.java

```diff
@@ -72,223 +75,246 @@
      * {@link org.hisp.dhis.common.IdentifiableObjectManager#get(Class, String)}.
      *
      * @param uid a UID of an object of unknown type
-     * @return The {@link IdentifiableObject} with the given UID or null if no
-     *         such object exists
+     * @return The {@link IdentifiableObject} with the given UID
      */
-    IdentifiableObject find( String uid );
+    @NotNull
```

Comment:
This is just Java being Java.
We are returning an Optional but technically we still need to be sure that it is not null :(

Reply:
Yes, `Optional` is not privileged in any way and will require the annotation.